### PR TITLE
Implement @alpha annotation

### PR DESCRIPTION
**Pull request by odersky**
The second half of the implementation of https://dotty.epfl.ch/docs/reference/changed-features/operators.html.

### CODE REVIEW

File: tests/pos/alpha.scala [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import annotation.alpha
+
+object Test {
+
+  def foo() = 1
+
+  @alpha("bar") def foo(x: Int) = 2
```
Comment (sjrd) [inline tests/pos/alpha.scala]: I'd like to see a pos test like the following:
```scala
def foo(x: Any): Int = 1
@alpha("bar") def foo[A](x: A): Int = 2
```
Without `@alpha`, those overloads would clash after erasure. But with `@alpha`, they should be fine.

File: tests/run/alpha/Test_2.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+package alpha;
+
+public class Test_2 {
+
+  public static void main(String[] args) {
+    Alpha<String> a = new Bar();
+    assert a.foo() == 1;
+    assert a.bar("a").equals("aa");
+    Alpha<String> aa = a.append(a);
```
Comment (sjrd) [inline tests/run/alpha/Test_2.java]: I'd also like to see a test where Java extends a class with `@alpha` methods and overrides them.
More complicated, what happens when Scala with `@alpha` overrides Java methods.

Even more tricky: what about:
```scala
// Scala
class A {
  @alpha("bar") def foo(): Int = 1
}
```
```java
// Java
public class B extends A {
  @Override
  public int bar() {
    return 2;
  }
}
```
```scala
// Scala
class C extends B {
  // is this correct?
  @alpha("bar") override def foo(): Int = 3

  // what about this?
  override def bar(): Int = 4
}
```

File: compiler/src/dotty/tools/dotc/transform/ElimErasedValueType.scala [hunk 3]
```diff
@@ -95,7 +96,12 @@ class ElimErasedValueType extends MiniPhase with InfoTransformer { thisPhase =>
         (sym1.owner.derivesFrom(defn.PolyFunctionClass) ||
          sym2.owner.derivesFrom(defn.PolyFunctionClass))
 
-      if (!info1.matchesLoosely(info2) && !bothPolyApply)
+      if (sym1.name != sym2.name && 
+          !(sym1.name.is(SuperAccessorName) && sym2.name.is(SuperAccessorName))
+            // super-accessors start as private, and their expanded name can clash after
+            // erasure. TODO: Verify that this is OK.
+          ||
```
Comment (sjrd) [inline compiler/src/dotty/tools/dotc/transform/ElimErasedValueType.scala]: This complex condition is not very readable. Especially with the `&&` and `||` seemingly at the same level, and the large blocks. Consider:
```scala
// super-accessors start as private, and their expanded name can clash after
// erasure. TODO: Verify that this is OK.
def bothSuperAccessors =
  sym1.name.is(SuperAccessorName) && sym2.name.is(SuperAccessorName)

if ((sym1.name != sym2.name && !bothSuperAccessors) ||
    (!info1.matchesLoosely(info2) && !bothPolyApply))
  ctx.error(DoubleDefinition(sym1, sym2, root), root.sourcePos)
```

File: tests/run/alpha/Test_2.java (hunk 2 above)
Comment (odersky) [reply tests/run/alpha/Test_2.java]: A Scala method with @alpha cannot possibly override a Java method. Overriding before and after erasure must be the same. I explore three possible scenarios in a new test neg/alpha-override.

**Pull request merged**